spec/models/spree/calculator/flat_percent_item_total_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'spec_helper'
+
+describe Spree::Calculator::FlatPercentItemTotal, type: :model do
+  let!(:promotion_two) do
+    promotion = Spree::Promotion.create!(name: "10% discount", apply_automatically: true)
+
+    calculator = Spree::Calculator::FlatPercentItemTotal.new
+    calculator.preferred_flat_percent = 10
+
+    action = Spree::Promotion::Actions::CreateAdjustment.create
+    action.calculator = calculator
+    action.save
+
+    promotion.actions << action
+  end
+
+  let(:gift_card) { create(:virtual_gift_card) }
+  let(:order) { create(:order_with_line_items, state: 'cart', line_items: [gift_card.line_item]) }
+
+  context "compute" do
+    it "should exclude gift cards from calculations" do
```

Comment:
[Better specs has something to say about shoulds](http://www.betterspecs.org/#should)